tools/eslint-rules/unescaped-regexp-dot.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @fileoverview Look for unescaped "literal" dots in regular expressions
+ * @author Brian White
+ */
+'use strict';
+
+const path = require('path');
+const utilsPath = path.join(__dirname, '..', 'eslint', 'lib', 'ast-utils.js');
+const astUtils = require(utilsPath);
+const getLocationFromRangeIndex = astUtils.getLocationFromRangeIndex;
+const getRangeIndexFromLocation = astUtils.getRangeIndexFromLocation;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+  const sourceCode = context.getSourceCode();
+  const regexpStack = [];
+  var regexpBuffer = [];
+  var inRegExp = false;
+
+  function report(node, startOffset) {
+    context.report({
+      node,
+      loc: getLocationFromRangeIndex(
+        sourceCode,
+        getRangeIndexFromLocation(sourceCode, node.loc.start) + startOffset
+      ),
+      message: 'Unescaped dot character in regular expression'
+    });
+  }
+
+  function checkRegExp(nodes) {
+    var escaping = false;
+    var inCharClass = false;
+    for (var n = 0; n < nodes.length; ++n) {
+      const pair = nodes[n];
+      const node = pair[0];
+      const str = pair[1];
+      for (var i = 0; i < str.length; ++i) {
+        switch (str[i]) {
+          case '[':
+            if (!escaping)
+              inCharClass = true;
+            else
+              escaping = false;
+            break;
+          case ']':
+            if (!escaping) {
+              if (inCharClass)
+                inCharClass = false;
+            } else {
+              escaping = false;
+            }
+            break;
+          case '\\':
+            escaping = !escaping;
+            break;
+          case '.':
+            if (!escaping) {
+              if (!inCharClass && (i + 1) < str.length) {
```

Comment:
Would we want to report the node when `i + 1 === str.length`? This would cover the case where the `.` is at the end of the regex, e.g.

```js
assert.throws(func, /this error message ends with a dot./);
```